src/internal/clusterstate/v2.8.0/pfsdb.go

```diff
@@ -308,19 +336,175 @@ func createNotifyCommitsTrigger(ctx context.Context, tx *pachsql.Tx) error {
 
 }
 
-// Migrate commits from collections.commits to pfs.commits
 func migrateCommitSchema(ctx context.Context, env migrations.Env) error {
 	if err := alterCommitsTable(ctx, env.Tx); err != nil {
 		return err
 	}
 	if err := createCommitAncestryTable(ctx, env.Tx); err != nil {
 		return err
 	}
-	// todo(fahad): migrate commits
-	// todo(fahad): call createNotifyCommitsTrigger() once migration is complete
 	return nil
 }
 
+type commitCollection struct {
+	v2_7_0.CollectionRecord
+	IntID uint64 `db:"int_id"`
+}
+
+func migrateCommitsFromCollections(ctx context.Context, env migrations.Env) error {
+	count := struct {
+		Collections uint64 `db:"col_count"`
+		Commits     uint64 `db:"commits_count"`
+	}{}
+	if err := env.Tx.GetContext(ctx, &count, `SELECT count(commit.int_id) AS commits_count, count(col.key) 
+    	AS col_count FROM pfs.commits commit LEFT JOIN collections.commits col on commit.commit_id = col.key;`); err != nil {
+		return errors.Wrap(err, "counting rows in collections.commits")
+	}
+	if count.Collections != count.Commits {
+		return errors.New(fmt.Sprintf("collections.commits has %d rows while pfs.commits has %d rows",
+			count.Collections, count.Commits))
+	}
+	if count.Collections == 0 {
+		return nil
+	}
+	pageSize := uint64(100)
```

Comment:
I think 100 per page might be a bit low, and could introduce too much IO overhead as the number of pages grow on the order of `O(commits_count/100)`.

```suggestion
	pageSize := uint64(1000)
```